Applications/Spire/Include/Spire/Ui/DecimalInputWidget.hpp

```diff
@@ -0,0 +1,53 @@
+#ifndef SPIRE_DECIMAL_INPUT_WIDGET_HPP
+#define SPIRE_DECIMAL_INPUT_WIDGET_HPP
+#include <QDoubleSpinBox>
+#include "Spire/Spire/Spire.hpp"
+
+namespace Spire {
+
+  //! Represents a widget for displaying and modifying decimal numbers.
+  class DecimalInputWidget : public QDoubleSpinBox {
+    public:
+
+      //! Signals a user interaction with the value.
+      using ValueSignal = Signal<void (double value)>;
+
+      //! Constructs a DecimalInputWidget.
+      /*!
+        \param value The initial value to display.
+        \param parent The parent widget.
+      */
+      explicit DecimalInputWidget(double value, QWidget* parent = nullptr);
+
+      //! Connects a slot to the value change signal.
+      boost::signals2::connection connect_change_signal(
```

Comment:
You may be able to get rid of these signals and stick to using ```QDoubleSpinBox```'s signals. All this widget does is style a ```QDoubleSpinBox``` correct?

Reply:
I might be able to get rid of the signals, but there's a few changes related to editing that go beyond just a style change.

Reply:
Okay, keep all the logic then, remove the signals.